staging/kube-oidc-proxy/templates/deployment.yaml

```diff
@@ -51,6 +51,8 @@ spec:
           - "--oidc-username-claim=$(OIDC_USERNAME_CLAIM)"
           {{- if or .Values.oidc.caPEM .Values.oidc.caSecretName }}
           - "--oidc-ca-file=/etc/oidc/oidc-ca.pem"
+          {{- else if and .Values.oidc.caKubernetesDefault }}
```

Comment:
Do you also need to check for `oidc.caCertPemHostPath` here and set `--oidc-ca-file` accordingly? I think we should define the precedence order:
1. caPEM or caSecretName
2. caCertPemHostPath
3. caKubernetesDefault
4. system default in the image

lower the number means higher the priority. We should document this in the values file.

Also, no need for `and` here.